File: src/solution_java/medium/_334IncreasingTripletSubsequence.java
package solution_java.medium;

public class _334IncreasingTripletSubsequence {
    public class Solution {
        public boolean increasingTriplet(int[] nums) {
            int first = Integer.MAX_VALUE;
            int second = Integer.MAX_VALUE;
            for (int i = 0; i < nums.length; i++) {
                int item = nums[i];
                if (item <= first) {
                    first = item;
                } else if (item <= second) {
                    second = item;
                } else {
                    return true;
                }
            }
            return false;
        }
    }
}

File: src/medium/_334IncreasingTripletSubsequence.kt
package medium

class _334IncreasingTripletSubsequence {
    class Solution {
        fun increasingTriplet(nums: IntArray): Boolean {
            var small = Int.MAX_VALUE
            var middle = Int.MAX_VALUE
            for (i in nums.indices) {
                val num = nums[i]
                if (num <= small) {
                    small = num
                } else if (num <= middle) {
                    middle = num
                } else {
                    return true
                }
            }
            return false
        }
    }
}